### ISSUE TITLE
`nextrng` not checkpointed, consider using `fold_in(config.seed, step)`

### ISSUE THREAD

**Issue by reinerp** [google/maxtext#276]
Looks like the `nextrng` value is not saved to the checkpoint: https://github.com/google/maxtext/blob/bab4ee78fdeb7a87e0ad01d3f9fa822229ddea07/MaxText/train.py#L285

It doesn't matter in most cases since most people don't use dropout or stochastic rounding. But in cases where it does matter, it's cleaner to generate the RNG for a training step using `jax.random.fold_in(config.seed, state.step)`. This way, no checkpointing is required, and there's also some other side advantages listed in https://twitter.com/cgarciae88/status/1615022554992738315.

**Comment by rwitten** [comment]
We will follow the suggestion here.

**Comment by rwitten** [comment]
done!
